core/client/fs/src/main/java/alluxio/client/PositionRead.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client;
+
+import alluxio.client.file.cache.store.ByteArrayTargetBuffer;
+import alluxio.client.file.cache.store.ByteBufferTargetBuffer;
+import alluxio.client.file.cache.store.NettyBufTargetBuffer;
+import alluxio.client.file.cache.store.PageReadTargetBuffer;
+import io.netty.buffer.ByteBuf;
+
+import java.nio.ByteBuffer;
+
+public interface PositionRead {
+  default  int read(PathIdentifier fileId, long position, byte[] buffer, int offset, int size) {
+    return read(fileId, position, new ByteArrayTargetBuffer(buffer, offset), size);
+  }
+
+  default  int read(PathIdentifier fileId, long position, ByteBuffer buffer, int size) {
+    return read(fileId, position, new ByteBufferTargetBuffer(buffer), size);
+  }
+
+  default  int read(PathIdentifier fileId, long position, ByteBuf buffer, int size) {
+    return read(fileId, position, new NettyBufTargetBuffer(buffer), size);
+  }
+
+  int read(PathIdentifier fileId, long position, PageReadTargetBuffer buffer, int size);
```

Comment:
Does the PageReadTargetBuffer suits for all cases? or does it need to be modify to support other cases e.g. read from UFS, read from netty